### DISCUSSION TITLE
Installing Obico does not install Janus

### DISCUSSION THREAD

**Discussion by VashBaldeus** [Guilouz/Creality-Helper-Script-Wiki#716] [Issues]
**UPDATE:**
I have the following config for moonraker-obico.cfg:
```cfg
[server]
url = https://app.obico.io
auth_token = TOKEN_HERE

[moonraker]
host = 127.0.0.1
port = 7125

[webcam]
disable_video_streaming = False
snapshot_url = http://127.0.0.1:4408/webcam/?action=snapshot
stream_url = http://127.0.0.1:4408/webcam/?action=stream
target_fps = 15

[logging]
path = /usr/data/printer_data/logs/moonraker-obico.log

[tunnel]


[misc]
sentry_opt: in
```
This config allowed the Janus error to go away and now the camera works.

===================================================

I have used the script to install Moonraker-Obico and connected it to Obico. Everything works except the camera. On the ucp of Obico I saw the following error: `Janus not found or not configured correctly. Click the link for troubleshooting guide.` (link https://www.obico.io/docs/user-guides/webcam-install-janus/) 
In their guide they say you need to run `sudo apt-get install -y janus` 
But due to the nature of the Linux version you can't install it easily. This began after Creality pushed 1.3.3.36 update to my K1 Max machine.

**Comment by JoeDickerson86** [comment]
Same, no Janus. Im still newish to linux so I cant figure out how to install it on a buildroot system.

**Comment by Tareku99** [comment]
Any updates on this? Cannot install janus and also cannot get the service to actually run in the klipper UI.

Solutions would be appreciated

**Reply by VashBaldeus** [reply]
> Any updates on this? Cannot install janus and also cannot get the service to actually run in the klipper UI.
> 
> Solutions would be appreciated

I've done the following in the Obico config:
```cfg
[server]
url = https://app.obico.io
auth_token = TOKEN_HERE

[moonraker]
host = 127.0.0.1
port = 7125

[webcam]
disable_video_streaming = False
snapshot_url = http://127.0.0.1:4408/webcam/?action=snapshot
stream_url = http://127.0.0.1:4408/webcam/?action=stream
target_fps = 15

[logging]
path = /usr/data/printer_data/logs/moonraker-obico.log

[tunnel]


[misc]
sentry_opt: in
```
As you can see I directed the photo & stream to the direct link of the camera. For me with this config the error vanished and it works.

**Reply by Tareku99** [reply]
Late reply but sadly this did not work either. Anyone have another solution?

**Reply by JoeDickerson86** [reply]
> Late reply but sadly this did not work either. Anyone have another solution?

So, I did a work around. Setup a cheap orange pi zero plugged into the USB port. Using the pi for the Janus server. I now have control with obico, and a working webcam

**Reply by Tareku99** [reply]
You created a Janus server? My current setup is in unraid (obico-server is running there) and I am trying to monitor a K1C running rooted klipper

EDIT: created an [issue on the docker image](https://github.com/imagegenius/docker-obico/issues/31) itself as well

**Reply by JoeDickerson86** [reply]
> You created a Janus server? My current setup is in unraid (obico-server is running there) and I am trying to monitor a K1C running rooted klipper
> 
> EDIT: created an [issue on the docker image](https://github.com/imagegenius/docker-obico/issues/31) itself as well

What I did, which is kind of a dirty way of doing it, I flashed the pi with a light version of Linux. Ran kiauh. Installed Klipper and obico. The Klipper instance does nothing, but it allowed me to get a proper obico install with Janus. If I was smarter with Linux I could probably skip the Klipper install part. Simply took the webcam address from the k1c, and put it into the pi obico config. It's a dirty method, but it's been working flawless for about a month now.

**Reply by Tareku99** [reply]
Ah I see, so in theory if I can get janus to be installed in the docker container that runs obico-server.. and change the obico config file, it should work? Do you know where the obico config file is located (on the pi)?

**Reply by JoeDickerson86** [reply]
> Ah I see, so in theory if I can get janus to be installed in the docker container that runs obico-server.. and change the obico config file, it should work? Do you know where the obico config file is located (on the pi)?

I don't remember. I'm not at home to check the discord. I had posted on the obico discord back when I was doing this. Can't remember if it was the Moonraker obico config file but I think that was it.

**Comment by rickythewoof** [comment]
hello, unfortunately adding and correctly setting the snapshot/stream url did not help me. I still have the same issue of a black video screen, and the Janus error. I am running obico over a proxy with websocket support and without forcing SSL 
also, I don't have any entry on my moonraker-obico log, even with DEBUG flag set
